src/extensions/custom-view/bootstrap-table-custom-view.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @author: Dustin Utecht
+ */
+
+const Utils = $.fn.bootstrapTable.utils
+
+$.extend($.fn.bootstrapTable.defaults, {
+  customView: false,
+  showCustomView: false,
+  showCustomViewButton: false
+})
+
+$.extend($.fn.bootstrapTable.defaults.icons, {
+  customView: {
+    bootstrap3: 'glyphicon glyphicon-eye-open',
+    bootstrap4: 'fa fa-eye',
+    semantic: 'fa fa-eye',
+    foundation: 'fa fa-eye',
+    bulma: 'fa fa-eye',
+    materialize: 'remove_red_eye'
+  }[$.fn.bootstrapTable.theme] || 'fa-eye'
+})
+
+$.extend($.fn.bootstrapTable.defaults, {
+  onCustomViewPostBody () {
+    return false
+  },
+  onCustomViewPreBody () {
+    return false
+  }
+})
+
+$.extend($.fn.bootstrapTable.locales, {
+  formatToggleCustomView () {
+    return 'Toggle customview'
+  }
+})
+$.extend($.fn.bootstrapTable.defaults, $.fn.bootstrapTable.locales)
+
+$.extend($.fn.bootstrapTable.Constructor.EVENTS, {
+  'custom-view-post-body.bs.table': 'onCustomViewPostBody',
+  'custom-view-pre-body.bs.table': 'onCustomViewPreBody'
+})
+
+$.BootstrapTable = class extends $.BootstrapTable {
+
+  init () {
+    this.showCustomView = this.options.showCustomView || true
+    super.init()
+  }
+
+  initToolbar (...args) {
+    super.initToolbar(...args)
+
+    if (this.options.customView && this.options.showCustomViewButton) {
+      const $btnGroup = this.$toolbar.find('>.' + this.constants.classes.buttonsGroup.split(' ').join('.')).first()
+      let $btnToggleCustomView = $btnGroup.find('.toggle-custom-view')
+
+      if (!$btnToggleCustomView.length) {
+        $btnToggleCustomView = $(`
+          <button class="toggle-custom-view ${this.constants.buttonsClass}
+          ${this.options.autoRefreshStatus ? ` ${this.constants.classes.buttonActive}` : ''}"
+          type="button" title="${this.options.formatToggleCustomView()}">
+          ${this.options.showButtonIcons ? Utils.sprintf(this.constants.html.icon, this.options.iconsPrefix, this.options.icons.customView) : ''}
+          ${this.options.showButtonText ? this.options.formatToggleCustomView() : ''}
+          </button>
+        `).appendTo($btnGroup)
+
+        $btnToggleCustomView.on('click', $.proxy(this.toggleCustomView, this))
+      }
+    }
+  }
+
+  initBody () {
+    super.initBody()
+    const $table = this.$el
+    const $customViewContainer = this.$container.find('.fixed-table-custom-view')
+
+    $table.hide()
```

Comment:
Let's hide the table only when the customView is false (or $.noop) and showCustomView is false. Because you're hide it and them show it after customView and showCustomView evaluation.